src/orders-and-subscriptions/OrdersAndSubscriptionsPage.jsx

```diff
@@ -1,18 +1,48 @@
 import React, { useEffect } from 'react';
-import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { useDispatch, useSelector } from 'react-redux';
+import { getConfig } from '@edx/frontend-platform';
+import { sendTrackEvent } from '@edx/frontend-platform/analytics';
+import { FormattedMessage, useIntl } from '@edx/frontend-platform/i18n';
 
-import Subscriptions from '../subscriptions';
-import OrderHistory from '../order-history';
+import Subscriptions, { fetchSubscriptions } from '../subscriptions';
+import OrderHistory, { fetchOrders } from '../order-history';
+import { loadingSelector } from './selectors';
+import { PageLoading } from '../common';
 
 const OrdersAndSubscriptionsPage = () => {
-  // TODO: get from waffle-flag
-  const isB2CSubsEnabled = true;
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const loading = useSelector(loadingSelector);
+  const isB2CSubsEnabled = JSON.parse(
+    getConfig().ENABLE_B2C_SUBSCRIPTIONS ?? 'false',
+  );
 
   useEffect(() => {
     if (isB2CSubsEnabled) {
       document.title = 'Orders and Subscriptions | edX';
+      sendTrackEvent('edx.bi.user.subscription.order-page.viewed');
+      dispatch(fetchSubscriptions());
     }
-  }, [isB2CSubsEnabled]);
+    dispatch(fetchOrders(1));
```

Comment:
Could you add the original TODO back? I know it's old, but I believe Purchase squad hasn't addressed this yet 😅
`// TODO: We should fetch based on the route (ex: /orders/list/page/1)`

Reply:
Updated with a modified comment (changed example to orders/?ordersPage=1)

Reasoning: since we have two sections on the page now only a single page argument in url would be ambiguous, especially after we add pagination to subscriptions section also in the future, hence it should be more appropriate to have a queryparam specifying whichPage are you referring to